client/src/home/components/Referral.js

```diff
@@ -109,7 +109,7 @@ export default function Referral() {
               <TextField
                 variant="outlined"
                 id="referralLink"
-                value="https://www.messenger.com/join/363274"
+                value={`${window.location.protocol}//${window.location.hostname}:${window.location.port}/join/${props.id}`}
```

Comment:
this may not work when you launch this app under an actual domain because `window.location.port` may be an empty string. and you might end up something like like `http://yourdomain.com:`

one suggestion i have would be to identify your environment. 

https://create-react-app.dev/docs/adding-custom-environment-variables#adding-temporary-environment-variables-in-your-shell

> There is also a special built-in environment variable called NODE_ENV. You can read it from process.env.NODE_ENV. When you run npm start, it is always equal to 'development', when you run npm test it is always equal to 'test', and when you run npm run build to make a production bundle, it is always equal to 'production'. You cannot override NODE_ENV manually. This prevents developers from accidentally deploying a slow development build to production.

perhaps you can send the environment as a prop (you can do this on a more global level, this wouldn't hurt but since this is really the only place you need it, local is fine too). and do something like 
```
if (env === 'development') {
  // dev env
  url = 'http://localhost:3000';
} else if (env === 'production') {
  url = 'http://messenger.com';
}
```

another benefit to this way is that you can perhaps use a url shortener (redirects back to the app) so that it takes up less characters, and more "shareable" through text. with this method, you have the ability to change your referral url to whatever you want without depending on the app's host :)